### ISSUE TITLE
[filer.sync] new file sync inconsistency

### ISSUE THREAD

**Issue by creeew** [chrislusf/seaweedfs#3326]
**Describe the bug**
Adding the same name file with different content at the same time in two different syncing clusters. We will find they exchange the file content.
e.g.
There are two clusters naming ClusterA and ClusterB, and using filer.sync to sync files to each other.
ClusterA adding a new file test.txt with content "aaa" and ClusterB adding a new file test.txt with content "bbb" at the same time. And then ClusterA have a file test.txt with content "bbb" and ClusterB have a file test.txt with content "aaa". The file's content exchanged.
Maybe the above senario is kinda hard to reproduce, we could assume filer.sync is broken. Cluster A and ClusterB adding the same file with different content at that time. We can find the file content exchanged when the filer.sync recover.

**Expected behavior**
Adding same file at the same time in different syncing cluster still have sequence. We could have guaranteeing consistency if the later change is our eventual change.
ClusterA adding "aaa" and ClusterB adding "bbb", ClusterB's change is slightly later than ClusterA. Then the file test.txt content is "bbb" in ClusterA and ClusterB.

**Additional context**
https://github.com/chrislusf/seaweedfs/blob/1db012485f996df0615a1d6a7ece9faace2e536c/weed/replication/sink/filersink/filer_sink.go#L114
We can compare the file's mtime. Skipping the change if newEntry's mtime earlier than existing one.

**Comment by creeew** [comment]
We could optimize filer.sync's replay if we add mtime compare when CreateEntry.
ClusterA and ClusterB are using filer.sync to sync files to each other.
ClusterA create file a.txt, and writing a large text to a.txt (ClusterA would create two metaData: 1. create file, 2. update file)
ClusterB sync the file creation and replay the large file content. And store the ClusterA's sync offset.
But in somehow weird situation, clusterA's sync offset is missing in CluserB's filer store. ClusterB have to sync the file a.txt again but ClusterB literally have the a.txt. ClusterB need to create a.txt and replay the large file content again.
There are nothing to do if we compare the mtime when clusterB received the a.txt create event, cause ClusterB know it synced the change already.

**Comment by chrislusf** [comment]
there are no guarantee the two clusters have the same time.

**Comment by creeew** [comment]
> there are no guarantee the two clusters have the same time.

Yes, but in filer.syn broken situation, ClusterA add the file and then CluserB add the same name file with different content. File's content exchanged when the filer.sync recover.
I thought compare the mtime when creatEntry(filer_sink) is the easiest way to deal this situation.

**Comment by creeew** [comment]
filer.sync may not have consistency guarantee if having network latency in syncing clusters.

e.g.
ClusterA and ClusterB syncing file hello.txt and there are network latency in syncing clusters.
ClusterA append a new line to hello.txt
and then ClusterB delete the file hello.txt before ClusterB receive ClusterA's append eventNotification cause the network latency.
ClusterA receive ClusterB's delete eventNotification, ClusterA notice ClusterB's eventNotification's mtime is later than existing one, deleted the file in ClusterA.
ClusterB received ClusterA's append eventNotification, just create hello.txt with full content.
In a word, they exchanged file's content.

We cannot tell received eventNotification is later or not since file already deleted. Maybe we could store deleted file metadata in filer store with ttl to store deleted file mtime.
